model/textparse/protobufparse_test.go

```diff
@@ -892,3 +896,665 @@ metric: <
+			v: 0.000828,
+			lset: labels.FromStrings(
+				"__name__", "test_histogram2_sum",
+			),
+		},
+		{ // 49
+			m: "test_histogram2_bucket\xffle\xff-0.00048",
+			v: 2,
+			lset: labels.FromStrings(
+				"__name__", "test_histogram2_bucket",
+				"le", "-0.00048",
+			),
+		},
+		{ // 50
+			m: "test_histogram2_bucket\xffle\xff-0.00038",
+			v: 4,
+			lset: labels.FromStrings(
+				"__name__", "test_histogram2_bucket",
+				"le", "-0.00038",
+			),
+			e: []exemplar.Exemplar{
+				{Labels: labels.FromStrings("dummyID", "59727"), Value: -0.00038, HasTs: true, Ts: 1625851153146},
+			},
+		},
+		{ // 51
+			m: "test_histogram2_bucket\xffle\xff1.0",
+			v: 16,
+			lset: labels.FromStrings(
+				"__name__", "test_histogram2_bucket",
+				"le", "1.0",
+			),
+			e: []exemplar.Exemplar{
+				{Labels: labels.FromStrings("dummyID", "5617"), Value: -0.000295, HasTs: false},
+			},
+		},
+		{ // 52
+			m: "test_histogram2_bucket\xffle\xff+Inf",
+			v: 175,
+			lset: labels.FromStrings(
+				"__name__", "test_histogram2_bucket",
+				"le", "+Inf",
+			),
+		},
+		{ // 53
+			m:    "rpc_durations_seconds",
+			help: "RPC latency distributions.",
+		},
+		{ // 54
+			m:   "rpc_durations_seconds",
+			typ: MetricTypeSummary,
+		},
+		{ // 55
+			m: "rpc_durations_seconds_count\xffservice\xffexponential",
+			v: 262,
+			lset: labels.FromStrings(
+				"__name__", "rpc_durations_seconds_count",
+				"service", "exponential",
+			),
+		},
+		{ // 56
+			m: "rpc_durations_seconds_sum\xffservice\xffexponential",
+			v: 0.00025551262820703587,
+			lset: labels.FromStrings(
+				"__name__", "rpc_durations_seconds_sum",
+				"service", "exponential",
+			),
+		},
+		{ // 57
+			m: "rpc_durations_seconds\xffservice\xffexponential\xffquantile\xff0.5",
+			v: 6.442786329648548e-07,
+			lset: labels.FromStrings(
+				"__name__", "rpc_durations_seconds",
+				"quantile", "0.5",
+				"service", "exponential",
+			),
+		},
+		{ // 58
+			m: "rpc_durations_seconds\xffservice\xffexponential\xffquantile\xff0.9",
+			v: 1.9435742936658396e-06,
+			lset: labels.FromStrings(
+				"__name__", "rpc_durations_seconds",
+				"quantile", "0.9",
+				"service", "exponential",
+			),
+		},
+		{ // 59
+			m: "rpc_durations_seconds\xffservice\xffexponential\xffquantile\xff0.99",
+			v: 4.0471608667037015e-06,
+			lset: labels.FromStrings(
+				"__name__", "rpc_durations_seconds",
+				"quantile", "0.99",
+				"service", "exponential",
+			),
+		},
+		{ // 60
+			m:    "without_quantiles",
+			help: "A summary without quantiles.",
+		},
+		{ // 61
+			m:   "without_quantiles",
+			typ: MetricTypeSummary,
+		},
+		{ // 62
+			m: "without_quantiles_count",
+			v: 42,
+			lset: labels.FromStrings(
+				"__name__", "without_quantiles_count",
+			),
+		},
+		{ // 63
+			m: "without_quantiles_sum",
+			v: 1.234,
+			lset: labels.FromStrings(
+				"__name__", "without_quantiles_sum",
+			),
+		},
+	}
+
+	p := NewProtobufParser(inputBuf.Bytes(), true)
+	i := 0
+
+	var res labels.Labels
+
+	for {
```

Comment:
LGTM. Nit: this part looks very similar to the code in `TestProtobufParse`, can it be reused?

Reply:
Good point. I had only looked at the first part of the test. Will make that DRY, too.

Reply:
I now made it a table-driven test after all.